### PR TITLE
Copia 11 i

### LINKED ISSUE #11: Sign Up Design
# REQUIREMENTS
This is for the Sign Up view that will bring the user the corresponding screen which will show the input texts to create an account by using his name, email and password, also bringing the possibility to Sign Up using google services and the corresponding checkbox to terms and subscription and finally the option to go to Login screen if an account exists.

# DESIGN
image

# Acceptance Criteria
List all required acceptance criteria
- [ ] The screen view and design must be like the one brought above.

### PR DESCRIPTION
**Pull request by alcamk**
# Description
- Complete Sign Up screen design
- Buttons, checkboxes and text fields with normal functionality
- Creation of checkbox component 
- Text with link option functionality 
- This is linked to #11 

 
 Examples:
- You can write in the text fields and check the color change of the border
- You can click the eye icon and check for password hidden functionality 
 
 # Testing
Go to Reservaciones/ folder and run $ npm install
Go to ios/ folder and run $ pod install
Go back to the project folder $ cd ..
Run $ npx react-native start
Run $ npx react-native run-ios for ios o $ npx react-native run-android for android 
  
  ## Screenshots 
captura3 png
captura6
captura5
captura1
captura2
captura4

### CODE REVIEW

**Review by JordiGN — DISMISSED** [review]
nice one adding the const for texts and colors

File: Reservaciones/components/TextField.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+TextField.propTypes = {
+  title: PropTypes.string,
+  Hfocus:  PropTypes.func,
+  HBlur:  PropTypes.func,
+  campo: PropTypes.string,
+  keyboard : PropTypes.bool,
+  errorLogin: PropTypes.string,
+  changeText: PropTypes.func
```
Comment (luisarmandodv) [inline Reservaciones/components/TextField.js]: `nameHandlerFocus` not defined as a PropType

File: Reservaciones/components/CheckboxCom.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      <Text style={{color: 'red'}}>{Texts.star}</Text>
+    </>
+    )
+}
+
+class  ComCheckbox extends React.Component {
+    constructor(props) {
+        super(props);
+
+      }
```
Comment (luisarmandodv) [inline Reservaciones/components/CheckboxCom.js]: pointless constructor

File: Reservaciones/components/CheckboxCom.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { View, Text } from 'react-native';
+import checkboxStyle from '../Styles/CheckboxStyle';
+import { Texts } from '../ContentText/TextsSignup';
+import {Colors} from '../Styles/Colors';
+import PropTypes from 'prop-types';
+
+
+const checked = () => {
+    console.log('Checked');
+}
```
Comment (luisarmandodv) [inline Reservaciones/components/CheckboxCom.js]: unused arrow function

File: Reservaciones/components/CheckboxCom.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import CheckBox from '@react-native-community/checkbox';
+import React, {useState} from 'react';
+import { View, Text } from 'react-native';
+import checkboxStyle from '../Styles/CheckboxStyle';
+import { Texts } from '../ContentText/TextsSignup';
+import {Colors} from '../Styles/Colors';
+import PropTypes from 'prop-types';
```
Comment (luisarmandodv) [inline Reservaciones/components/CheckboxCom.js]: global packages goes before local importing.

File: Reservaciones/components/Button.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import React, {Component} from 'react';
+import {View, Text, TouchableOpacity,Image} from 'react-native';
+import PropTypes from 'prop-types';
+import styleLogin from '../Styles/LoginStyle';
+
+
+class ButtonAction extends Component {
+  constructor(props) {
+    super(props);
+  }
```
Comment (luisarmandodv) [inline Reservaciones/components/Button.js]:
```suggestion
```

File: Reservaciones/components/Button.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        </TouchableOpacity>
+      </View>
+    );
+  }
+}
+
+ButtonAction.propTypes = {
+  imageRequired: PropTypes.bool,
+  title: PropTypes.string,
+  Press : PropTypes.func
```
Comment (luisarmandodv) [inline Reservaciones/components/Button.js]: `stateComponent` prop missing to be defined

File: Reservaciones/Styles/SignupStyle.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    },
+    TextTittle: {
+        color: Colors.blue,
+        fontSize: 22,
+        fontWeight: 'bold',
+        textAlign: 'left',
+        marginTop: 25,
+        marginBottom: 25,
+    },
+
```
Comment (luisarmandodv) [inline Reservaciones/Styles/SignupStyle.js]:
```suggestion
```

File: Reservaciones/Screens/Signup.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+                    <TextField
+                    title={Texts.pass}
+                    nameHandlerFocus={this.state.Vpassword}
+                    Hfocus={() => this.handlerFocus(Texts.Ppassword)}
+                    HBlur={() => this.handlerBlur(Texts.Ppassword, this.state.userPass)}
+                    campo= 'userPass'
+                    keyboard = {false}
+                    errorLogin=''
+                    changeText={this.handleChangeText}
+                  />
```
Comment (luisarmandodv) [inline Reservaciones/Screens/Signup.js]: indentation here is wrongly implemented,

File: Reservaciones/Screens/Signup.js [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        super(props); this.state = {
+             loading: false, 
+             isPasswordHidden: true, 
+             userEmail: '', 
+             userPass: '', 
+             errorLogin: '',
+             userFirstName: '',
+            };
+        }
+
```
Comment (luisarmandodv) [inline Reservaciones/Screens/Signup.js]: indentation here can be improved

File: Reservaciones/Screens/Signup.js [hunk 10]
```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'react';
+import {View, Text, TextInput} from 'react-native';
+import SignupStyle from '../Styles/SignupStyle';
+import PropTypes from 'prop-types';
```
Comment (luisarmandodv) [inline Reservaciones/Screens/Signup.js:5]: global importing goes before local importing

**Review by luisarmandodv — CHANGES_REQUESTED** [review]

File: Reservaciones/components/Button.js (hunk 6 above)
Comment (alcamk) [reply Reservaciones/components/Button.js]: @Ocosmee con you help me in this change pleas!

File: Reservaciones/Screens/Signup.js [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+import ButtonAction from '../components/Button';
+import ComCheckbox from '../components/CheckboxCom';
+import {Colors} from '../Styles/Colors';
+import styleLogin from '../Styles/LoginStyle';
+import SignupStyle from '../Styles/SignupStyle';
+
+
+
+
+
```
Comment (luisarmandodv) [inline Reservaciones/Screens/Signup.js]:
```suggestion
```

File: Reservaciones/components/Button.js (hunk 5 above)
Comment (luisarmandodv) [reply Reservaciones/components/Button.js]: what about this? @alcamk

File: Reservaciones/components/Button.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import React, {Component} from 'react';
+import {View, Text, TouchableOpacity,Image} from 'react-native';
+import PropTypes from 'prop-types';
+import styleLogin from '../Styles/LoginStyle';
+
```
Comment (luisarmandodv) [inline Reservaciones/components/Button.js:5]:
```suggestion
```

File: Reservaciones/Styles/SignupStyle.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+        width: '100%',
+    },
+    TextTittle: {
+        color: Colors.blue,
+        fontSize: 22,
+        fontWeight: 'bold',
+        textAlign: 'left',
+        marginTop: 25,
+        marginBottom: 25,
+    },
```
Comment (luisarmandodv) [inline Reservaciones/Styles/SignupStyle.js]: indentation has 4 spaces, but all the project I see it has 2 spaces.

File: Reservaciones/Styles/Colors.js [hunk 14]
```diff
@@ -0,0 +1,10 @@
+export const Colors = {
+    white: '#fff',
+    black: '#000',
+    blue: '#5C6EF8',
+    gray: '#B6B7BA',
+    red: '#FE1919',
+    darkGray: '#21252B',
+    transparent: '#00000040',
+  };
+
```
Comment (luisarmandodv) [inline Reservaciones/Styles/Colors.js:10]:
```suggestion
```

File: Reservaciones/Styles/CheckboxStyle.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+    },
+    item: {
+        marginTop: 60,
+        marginRight: 10,
+    },
+    txtChkbx: {
+        color: '#AFB2B7',
+        width: '100%',
+    }
+
```
Comment (luisarmandodv) [inline Reservaciones/Styles/CheckboxStyle.js]:
```suggestion
```

File: Reservaciones/Styles/CheckboxStyle.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+        flexDirection: "row",
+        backgroundColor:'blue'
+    },
+    item: {
+        marginTop: 60,
+        marginRight: 10,
+    },
+    txtChkbx: {
+        color: '#AFB2B7',
+        width: '100%',
```
Comment (luisarmandodv) [inline Reservaciones/Styles/CheckboxStyle.js]: indentation has 4 spaces, but all the project I see it has 2 spaces.

File: Reservaciones/Screens/Signup.js [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+    changeText: PropTypes.func,
+    imageRequired: PropTypes.bool,
+    Press: PropTypes.func,
+    innerText: PropTypes.string,
+    textRequire: PropTypes.func,
+    nameHandlerFocus: PropTypes.bool,
+    stateComponent: PropTypes.object
+  };
+
+export default Signup;
```
Comment (luisarmandodv) [inline Reservaciones/Screens/Signup.js:150]:
```suggestion
export default Signup;

```

File: Reservaciones/ContentText/TextsSignup.js [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+    and: 'and ',
+    pp: 'Privacy Policy.',
+    star: ' *',
+    checkbox2: 'Suscribe for select product updates.',
+    Ppassword: 'Vpassword',
+    Pfirstname: 'Vfirstname',
+    agreeCheckBox: 'agreeCheckBox',
+    subscribeCheckBox: 'subscribeCheckBox',
+  };
+
```
Comment (luisarmandodv) [inline Reservaciones/ContentText/TextsSignup.js:24]:
```suggestion
```

**Review by luisarmandodv — CHANGES_REQUESTED** [review]

**Review by luisarmandodv — APPROVED** [review]

**Pull request merged**